packages/platform-android/src/commands/runAndroid/tryLaunchEmulator.ts

```diff
@@ -0,0 +1,80 @@
+import execa from 'execa';
+import Adb from './adb';
+
+const getEmulators = () => {
+  try {
+    const emulatorsOutput = execa.sync('emulator', ['-list-avds']).stdout;
+    return emulatorsOutput.split('\n').filter(name => name !== '');
+  } catch {
+    return [];
+  }
+};
+
+const launchEmulator = async (emulatorName: string, adbPath: string) => {
+  return new Promise((resolve, reject) => {
+    const cp = execa('emulator', [`@${emulatorName}`], {
+      detached: true,
+      stdio: ['ignore', 'pipe', 'ignore'],
+    });
+
+    const timeout = 30;
```

Comment:
I think we can make it 15, which is already pretty long, but I can see there are slower environments to develop with. What do you think?
```suggestion
    const timeout = 15;
```

Reply:
I'm not sure about the timeouts here @thymikee . On My machine time to `emulator` prints
> emulator: INFO: boot completed
is 14.25 seconds. 15 may be to less. 
and to `adb devices` to return that device is online (or is a `device` actually) it takes about one second less, like 13 seconds. 
I was wondering if we could just check for adb devices in an interval using `adb.getDevices(adbPath)`. It could be one interval for both cold and snapshot boot.
Then we could clear interval once it is booted or after the delay

Reply:
Ah I assumed this is not the time to boot the emulator but to just start it. Do you think we need to wait for this event? I'd be cool with just starting it and failing only if it couldn't be started for some reason

Reply:
if the emulator is not booted but just started it will not be available to install an app. We can't assume that build takes some time (when emulator could boot properly) because in case of cached build it takes seconds. I think the best approach will be to ignore that event and in the interval, loop checks if there is a device on the list. It would simplify the code.

Reply:
Yea, I just realised that after I wrote the message. Do whatever is simpler and reliable